### PR TITLE
[JIT] Override print when python is present

### PR DESCRIPTION
**Pull request by jamesr66a**
This makes it so we can see the output of prim::Print in environments like iPython notebooks which override sys.stdout

### CODE REVIEW

File: torch/csrc/jit/print_handler.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+#include <atomic>
+#include <functional>
+#include <iostream>
+
+namespace torch {
+namespace jit {
+
+using PrintHandler = void (*)(const std::string&);
```
Comment (suo) [inline torch/csrc/jit/print_handler.h]: std::function?

File: torch/csrc/jit/init.cpp [hunk 2]
```diff
@@ -563,6 +566,16 @@ void initJITBindings(PyObject* module) {
...
   script::initJitScriptBindings(module);
+
+  setPrintHandler([](const std::string& str) {
+    AutoGIL ag;
+    // TODO: I can't get the below to work without crashing.
+    //       figure out how to make this work. Otherwise,
+    //       WriteStdout will truncate to 1000 bytes.
+    // auto _stdout = py::module::import("sys").attr("stdout");
+    // _stdout.attr("write")(_stdout, str);
+    PySys_WriteStdout("%s", str.c_str());
```
Comment (suo) [inline torch/csrc/jit/init.cpp]: can we not just call print through pybind?

File: torch/csrc/jit/print_handler.h (hunk 1 above)
Comment (jamesr66a) [reply torch/csrc/jit/print_handler.h]: nah brah

File: torch/csrc/jit/init.cpp (hunk 2 above)
Comment (jamesr66a) [reply torch/csrc/jit/init.cpp]: I can't get it to not complain that it expects bytes, not str, even if i explicitly cast to py::bytes. something seems broken there in pybind

**Review by suo — APPROVED** [review]

**Review by zdevito — APPROVED** [review]

**Pull request closed without merging**